build/generate-struct-arrays.js

```diff
@@ -0,0 +1,325 @@
+// @flow
+
+'use strict'; // eslint-disable-line strict
+
+require('flow-remove-types/register');
+
+const assert = require('assert');
+const fs = require('fs');
+const ejs = require('ejs');
+const util = require('../src/util/util');
+const spec = require('../src/style-spec/reference/v8');
+
+import type {ViewType} from '../src/util/struct_array';
+
+export type StructArrayTypeParameters = {
+    members: $ReadOnlyArray<{
+        name: string,
+        type: ViewType,
+        +components?: number,
+    }>,
+    alignment?: number
+};
+
+
+const structArrayLayoutJs = ejs.compile(fs.readFileSync('src/util/struct_array_layout.js.ejs', 'utf8'), {strict: true});
+const structArrayJs = ejs.compile(fs.readFileSync('src/util/struct_array.js.ejs', 'utf8'), {strict: true});
```

Comment:
What is the advantage of separating the layout and the rest of the struct array? Is there a lot of duplication that is avoided? What's the downside of the duplication?

Reply:
There's a fair amount of duplication.  One definite effect of the duplication is that it increases the code size; gzip probably makes this minimal for downloading, but the code still has to be parsed `1 + workerCount` times in the browser.  I also suspect that reducing the redundancy might make it a little easier for the optimizer to do its thing, but that's just a hypothesis.

Reply:
Maybe more importantly, though, is that this separation is a step towards eliminating "the rest of the struct array" part entirely for the array types where we don't actually make use of the struct accessors -- i.e., all the layout arrays and paint property arrays.

Reply:
Thanks, that makes sense!